comp/metadata/inventorysoftware/impl/inventorysoftware.go

```diff
@@ -0,0 +1,194 @@
+type sysProbeClientWrapper struct {
+	client *sysprobeclient.CheckClient
+}
+
+// GetCheck implements SysProbeClient.GetCheck by delegating to the wrapped client.
+// This method uses the generic GetCheck function to retrieve software inventory data
+// from the System Probe with proper type safety.
+func (w *sysProbeClientWrapper) GetCheck(module types.ModuleName) ([]software.Entry, error) {
+	return sysprobeclient.GetCheck[[]software.Entry](w.client, module)
+}
+
+// inventorySoftware is the implementation of the Component interface.
+// This struct holds the state and dependencies needed to collect and manage
+// software inventory data from the Windows system.
+type inventorySoftware struct {
+	util.InventoryPayload
+
+	// log provides logging capabilities for the component
+	log log.Component
+	// sysProbeClient is used to communicate with the System Probe for data collection
+	sysProbeClient SysProbeClient
+	// cachedInventory stores the most recently collected software inventory data
+	cachedInventory []software.Entry
+	// hostname identifies the system where the inventory was collected
+	hostname string
+	// enabled indicates whether software inventory collection is enabled in the configuration
+	enabled bool
+}
+
+// Requires defines the dependencies required by the inventory software component.
+// This struct defines all the required dependencies that must be provided
+// when creating a new inventory software component instance.
+type Requires struct {
+	// Log provides logging capabilities for the component
+	Log log.Component
+	// Config provides access to the agent configuration
+	Config config.Component
+	// Serializer is used to serialize and send data to the backend
+	Serializer serializer.MetricSerializer
+	// Hostname provides the hostname of the current system
+	Hostname hostnameinterface.Component
+}
+
+// Provides defines the output of the inventory software component.
+// This struct defines all the services and providers that the component
+// makes available to the rest of the system.
+type Provides struct {
+	// Comp is the main component interface for software inventory
+	Comp inventorysoftware.Component
+	// Provider is the metadata provider for software inventory data
+	Provider runnerimpl.Provider
+	// FlareProvider provides software inventory data for flare collection
+	FlareProvider flaretypes.Provider
+	// StatusHeaderProvider provides status information for the agent status page
+	StatusHeaderProvider status.HeaderInformationProvider
+	// Endpoint provides HTTP endpoint access to software inventory data
+	Endpoint api.AgentEndpointProvider
+}
+
+// New creates a new inventory software component with the default sysprobeclient
+func New(reqs Requires) (Provides, error) {
+	return NewWithClient(reqs, nil)
+}
+
+// NewWithClient creates a new inventory software component with a custom sysprobeclient
+func NewWithClient(reqs Requires, client SysProbeClient) (Provides, error) {
```

Comment:
To reduce package exposure, could we make this function private? Looking at the changes, we are only using them in the tests 😄